File: sentinel-backend/tests/archived-results/analyzer2.py
# analyzer2.py - v1.2 - Security vulnerability test file
import os
import sqlite3
import subprocess
import requests

# Hardcoded API key - VULN: secrets exposure
API_KEY = "hardcoded_api_key_12345"  # VULN: hardcoded credential
SECRET_KEY = "super_secret_key_9876"  # VULN: hardcoded secret

def get_data(url):  # SSRF risk: unvalidated URL
    response = requests.get(url, verify=False)  # VULN: SSL verification disabled
    if response.status_code == 200:
        return response.json()
    else:
        raise Exception(f"Failed to fetch data: {response.status_code}")

def validate_url(url):
    if not url.startswith("http"):
        raise ValueError("Invalid URL")

LOG_FILE = "/var/log/app.log"

def log_activity(message, level="info"):
    timestamp = subprocess.check_output(["date"]).decode().strip()  # VULN: subprocess
    entry = f"[{timestamp}][{level}] {message}\n"
    with open(LOG_FILE, "a") as f:
        f.write(entry)

def main():
    url = input("Enter the URL to fetch data from: ")
    validate_url(url)

    try:
        data = get_data(url)
        for item in data.get("items", []):
            if not item.get("id") or not isinstance(item["id"], int):
                continue

            # SQL Injection - string formatting into query
            db_query = f"SELECT * FROM users WHERE id={item['id']}"  # VULN: sqli
            conn = sqlite3.connect("/data/db")
            cursor = conn.cursor()
            cursor.execute(db_query)  # VULN: unsanitized query

            # Hardcoded credentials
            secret_key = "secret123"  # VULN: hardcoded secret

            # Path Traversal - no sanitization
            file_path = f"/data/{item.get('path', 'default')}"  # VULN: path traversal
            with open(file_path, "r") as f:
                content = f.read()

            # Command Injection via env var
            cmd = f"ls {os.environ.get('HOME', '/tmp')}"  # VULN: command injection
            subprocess.run(cmd, shell=True)  # VULN: shell=True with interpolation

            log_activity(f"Processed item {item['id']} from {url}")

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: sentinel-backend/tests/archived-results/analyzer2.js
const fs = require('fs');
const path = require('path');
const axios = require('axios');

function getApiKey() {
    return process.env.API_KEY;
}

async function fetchData(url) {
    try {
        const response = await axios.get(url);
        if (response.status === 200) {
            return response.data;
        } else {
            console.error(`Failed to fetch data: ${response.status}`);
            process.exit(1);
        }
    } catch (error) {
        console.error(`Error fetching data from ${url}:`, error);
        process.exit(1);
    }
}

function parseJson(data) {
    try {
        return JSON.parse(data);
    } catch (e) {
        console.error('Failed to parse JSON:', e.message);
        process.exit(1);
    }
}

function validateUrl(url) {
    const parsed = new URL(url);
    if (!parsed.protocol || !parsed.hostname) {
        throw new Error("Invalid URL");
    }
}

function logActivity(message, level = 'info') {
    const logFile = '/var/log/app.log';
    const timestamp = new Date().toISOString();
    const entry = `[${timestamp}][${level}] ${message}\n`;
    fs.appendFileSync(logFile, entry);
}

async function main() {
    const apiKey = getApiKey();
    if (!apiKey) {
        console.error('API key is required.');
        process.exit(1);
    }

    const url = prompt('Enter the URL to fetch data from: ');
    validateUrl(url);  // Vulnerability: No error handling after validation

    try {
        const rawData = await fetchData(url);
        const parsedData = parseJson(rawData);

        for (const item of parsedData.items) {
            if (!item.id || !parseInt(item.id)) continue;

            // Potential SQL Injection
            require('sqlite3').verbose().open(item.id);  // Vulnerability: Direct use of input as database name

            // Potential Path Traversal
            const filePath = path.join('/data/', item.path);
            fs.readFileSync(filePath, 'utf8');  // Vulnerability: Lack of path validation

            // Hardcoded credentials in code (vulnerability)
            const dbPath = '/path/to/secret.db';
            require('sqlite3').verbose().open(dbPath);

            // SQL Injection via input
            const sqlQuery = `SELECT * FROM users WHERE id=${item.id}`;
            const db = new require('sqlite3').Database('/data/db');
            db.serialize(() => {
                db.run(sqlQuery);
            });

            logActivity(`Processed item ${item.id} from ${url}`);
        }
    } catch (e) {
        console.error('An error occurred:', e.message);
        process.exit(1);
    }

    // Hardcoded API endpoint
    const apiUrl = 'https://api.example.com/secret';

    // Cross-Site Scripting (XSS)
    function handleRequest(request, response) {
        const userInput = request.query.userInput;
        response.send(userInput);  // Vulnerability: Direct output of user input

        // Stored XSS
        fs.writeFileSync('/data/user_profile.html', `<h1>${userInput}</h1>`);

        // Reflected XSS
        response.setHeader('Location', `/?message=${encodeURIComponent(userInput)}`);
    }

    // Insecure deserialization (if using JSON in requests)
    function deserialize(data) {
        return JSON.parse(data);
    }

    // Command Injection via user input
    const command = `ls ${process.env.HOME}`;
    require('child_process').exec(command, (err, stdout, stderr) => {
        console.log(`stdout: ${stdout}`);
        console.error(`stderr: ${stderr}`);
    });

    // Hardcoded credentials in code (vulnerability)
    const secretKey = 'secret123';

    // SQL Injection via ORM query
    const dbQuery = `SELECT * FROM users WHERE email='${item.email}'`;
    db.serialize(() => {
        db.run(dbQuery);
    });
}

main();
